Firestore/core/include/firebase/firestore/listener_registration.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_INCLUDE_FIREBASE_FIRESTORE_LISTENER_REGISTRATION_H_
+#define FIRESTORE_CORE_INCLUDE_FIREBASE_FIRESTORE_LISTENER_REGISTRATION_H_
+
+namespace firebase {
+namespace firestore {
+
+class FirestoreInternal;
+class ListenerRegistrationInternal;
+
+/** Represents a listener that can be removed by calling remove. */
+class ListenerRegistration {
+ public:
+  /**
+   * @brief Default constructor. This creates a no-op instance.
+   */
+  ListenerRegistration();
+
+  /**
+   * @brief Copy constructor. It's totally okay to copy ListenerRegistration
+   * instances.
+   *
+   * @param[in] registration ListenerRegistration to copy from.
```

Comment:
Personally, I'd remove this line. It's a copy ctor, so this kinda goes without saying.

Reply:
+1. I'd remove this comment and comments on move ctr/assign operator/etc. altogether, replacing them with two shorter comments: `Copying ListenerRegistration is fine` and `Moving ListenerRegistration is an efficient operation`.

Reply:
This is to make code consistent. All copy ctor in the Firestore public header has such param docs. I'd keep it to make the docs more complete (even this seems obvious and can be implicit) since eventually these will be part of the reference generated by doxygen.

Reply:
If it's just in fire*store*, then I'd argue we should fix it everywhere (not necessarily now, and certainly not as part of this import PR). I don't think we need to publish this as part of the doxygen docs as it's just stating the obvious. But if it's consistent across fire*base*, then that's not a battle I want to fight, so leaving it in would be fine by me in that situation.